File: filesystem/commands/mkdir/mkdir.ts
import { Command, CommandGenerator } from "../../core/command.ts";
import { Directory } from "../../core/directory.ts";
import { Environment } from "../../environment.ts";

class mkdir extends Command {
  public execute(input: string): void {
    this.env.cwd.addFsObject(new Directory(input, this.env.cwd));
  }
  
}

export const generator: CommandGenerator = (env: Environment) => {
  return new mkdir("mkdir", env);
}

File: filesystem/go-commands/mkdir/mkdir.go
package main

import (
	"github.com/DoktorLenz/kp-wise23-24/filesystem/go/core"
	"github.com/DoktorLenz/kp-wise23-24/filesystem/go/env"
)

func generator(environment *env.Environment) env.AbstractCommand {
	return env.AbstractCommand{
		Accessor:    "mkdir",
		Environment: environment,
		Execute: func(input string) error {
			environment.GetCwd().AddChild(core.NewDirectory(input, environment.GetCwd()).AbstractFsObject)
			return nil
		},
	}
}
